### ISSUE TITLE
ODE_MODULE_VERSION 59. This version of Node.js requires NODE_MODULE_VERSION 57. Please try re-compiling or re-installing

### ISSUE THREAD

**Issue by Mlhote1971** [eouia/MMM-AssistantMk2#1]
Hello
unable to operate the module
I do not know where my mistake is
if you can give me a hand I'm a taker

Thanks

`Starting MagicMirror: v2.4.1
Loading config ...
Loading module helpers ...
No helper found for module: alert.
No helper found for module: clock.
WARNING! Could not load config file. Starting with default configuration. Error found: Error: The module '/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/speaker/build/Release/binding.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 59. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
Loading module helpers ...
No helper found for module: alert.
No helper found for module: clock.
App threw an error during load
Error: The module '/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/speaker/build/Release/binding.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 59. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:671:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)
    at Function.Module._load (module.js:496:3)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at bindings (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/speaker/index.js:9:36)
Whoops! There was an uncaught exception...
Error: The module '/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/speaker/build/Release/binding.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 59. This version of Node.js requires
NODE_MODULE_VERSION 57. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:671:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)
    at Function.Module._load (module.js:496:3)
    at Module.require (module.js:586:17)
    at require (internal/module.js:11:18)
    at bindings (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/bindings/bindings.js:81:44)
    at Object.<anonymous> (/home/pi/MagicMirror/modules/MMM-AssistantMk2/node_modules/speaker/index.js:9:36)
MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
Launching application.`

**Comment by eouia** [comment]
First, I want to ask these;
- There was no error on your installation steps? your error messages are not for installation steps, just on running stage. So, you should be sure of your installation was cleared.
- Have you got Auth and made profiles already? Below solution could give you another problem if you are not finished all the steps before executing.
- If your installation and profiles getting were the success, you can check running in `Serveronly` mode. (`node serveronly`)

Your error caused by different node versions between `speaker` node-module and your current electron node version.

try this.
```shell
npm install --save-dev electron-rebuild
./node_modules/.bin/electron-rebuild
```

**Comment by Mlhote1971** [comment]
Great !
It's work
Thanks

Mat

**Comment by fermamitiprego** [comment]
Hallo everyone, I'm facing a similiar iusse. I've tried with the suggestion but, after the line "./node_modules/.bin/electron-rebuild" it stucks on "building module: grpc, completed: 0" and it doesn't go on. 
I am a very new user in linux word, so i am not really sure how provide you logs or other important informations about it. I've googled around before posting, either about the error (black screen, config file, etc) either about the "building module: grpc" (in that case, the only result is a japanese site in which google translator doesn't do its job.
I've to add that i had some problem during the installation of Hotword module, the part of snowboy module, and i've tried the workaround suggested in the guide. I don't remember if i had more error or warnings.

I am using an orange Pi PC with ubuntu xenial.
Thank you in advance!
